modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/exec/coercion/NumericArithmeticsExecutionTest.java

```diff
@@ -0,0 +1,618 @@
+import org.apache.ignite.internal.sql.engine.planner.datatypes.utils.TypePair;
+import org.apache.ignite.internal.sql.engine.planner.datatypes.utils.Types;
+import org.apache.ignite.internal.type.DecimalNativeType;
+import org.apache.ignite.internal.type.NativeTypes;
+import org.hamcrest.Matcher;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/** Check execution and return results for numeric arithmetics. */
+public class NumericArithmeticsExecutionTest extends BaseTypeCheckExecutionTest {
+    @ParameterizedTest
+    @MethodSource("plusMinusArgs")
+    public void sumOp(TypePair typePair, Matcher<Object> resultMatcher) throws Exception {
+        String sql = "SELECT c1 + c2 FROM t";
+        try (ClusterWrapper testCluster = testCluster(typePair, dataProviderReduced(typePair))) {
+            testCluster.process(sql, resultMatcher);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("plusMinusArgs")
+    public void subtractOp(TypePair typePair, Matcher<Object> resultMatcher) throws Exception {
+        String sql = "SELECT c1 - c2 FROM t";
+        try (ClusterWrapper testCluster = testCluster(typePair, dataProviderReduced(typePair))) {
+            testCluster.process(sql, resultMatcher);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("multArgs")
+    public void multOp(TypePair typePair, Matcher<Object> resultMatcher) throws Exception {
+        String sql = "SELECT c1 * c2 FROM t";
+        try (ClusterWrapper testCluster = testCluster(typePair, multDivDataProvider(typePair))) {
+            testCluster.process(sql, resultMatcher);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("divArgs")
+    public void divOp(TypePair typePair, Matcher<Object> resultMatcher) throws Exception {
+        String sql = "SELECT c1 / c2 FROM t";
+        Assumptions.assumeFalse(typePair.first() instanceof DecimalNativeType || typePair.second() instanceof DecimalNativeType,
+                "need to be fixed after: https://issues.apache.org/jira/browse/IGNITE-23171");
+
+        try (ClusterWrapper testCluster = testCluster(typePair, dataProviderStrict(typePair))) {
+            testCluster.process(sql, resultMatcher);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("moduloArgs")
+    public void moduloOp(TypePair typePair, Matcher<Object> resultMatcher) throws Exception {
+        String sql = "SELECT c1 % c2 FROM t";
+
+        try (ClusterWrapper testCluster = testCluster(typePair, dataProvider(typePair))) {
+            testCluster.process(sql, resultMatcher);
+        }
+    }
+
+    @Test
+    public void checkAllTypePairs() {
+        checkIncludesAllNumericTypePairs(plusMinusArgs());
+        checkIncludesAllNumericTypePairs(multArgs());
+        checkIncludesAllNumericTypePairs(divArgs());
```

Comment:
+ moduloArgs

Reply:
I remove not applicable types from there, thus it not passed. Modulo for example for 
select '2.1'::decimal(2,1) % '1.1'::float
is undefined, do you suggest to return "old" implementation where all pairs are scanned but results are filtered in operator ?

Reply:
fixed